vscode/src/chat/chat-view/ChatController.ts

```diff
@@ -698,29 +698,36 @@ export class ChatController implements vscode.Disposable, vscode.WebviewViewProv
         requestID,
         input,
         signal,
-        detectedIntent,
-        detectedIntentScores,
+        preDetectedIntent,
+        preDetectedIntentScores,
+        manuallySelectedIntent,
     }: {
         requestID: string
         input: string
         signal: AbortSignal
-        detectedIntent?: ChatMessage['intent'] | undefined | null
-        detectedIntentScores?: { intent: string; score: number }[] | undefined | null
+        preDetectedIntent?: ChatMessage['intent'] | undefined | null
+        preDetectedIntentScores?: { intent: string; score: number }[] | undefined | null
+        manuallySelectedIntent?: ChatMessage['intent'] | undefined | null
     }): Promise<{
         intent: ChatMessage['intent']
-        intentScores: { intent: string; score: number }[]
+        detectedIntent: ChatMessage['intent'] | undefined | null
+        detectedIntentScores: { intent: string; score: number }[]
     }> {
         if (!this.featureCodyExperimentalOneBox) {
-            return { intent: 'chat', intentScores: [] }
+            return { intent: 'chat', detectedIntent: null, detectedIntentScores: [] }
         }
 
-        // The `detectedIntent` comes from the webview, where it can either be manually set by the user
-        // using the dropdown UI or automatically pre-fetched for the input while the user is typing.
-        // If any such intent is already detected, we use that.
-        if (detectedIntent) {
+        // The `preDetectedIntent` and `manuallySelectedIntent` params come from the webview.
+        // If `manuallySelectedIntent` is set, this was a user override, and we should use it.
+        // If `preDetectedIntent` is set, the intent was automatically pre-fetched for the input,
+        // meaning we don't have to fetch it again.
+        const intent = manuallySelectedIntent ? manuallySelectedIntent : preDetectedIntent
```

Comment:
nit: especially since `manuallySelectedIntent` used to be type boolean, I think this makes it more clear that we only want to use the detected intent when `manuallySelectedIntent` is undefined or null

```suggestion
        const intent = manuallySelectedIntent ?? preDetectedIntent
```

Reply:
Oh nice, I'm still getting used to TS. This is great! :)